### DISCUSSION TITLE
[Branch protection] Disallow pushes to main without disabling force-merging PRs

### DISCUSSION THREAD

**Discussion by albertvaka** [community/community#12516] [Pull Requests]
Right now both things go together under the "Require a pull request before merging" branch protection option. 

However, it would be really useful to separate them, to protect main from pushing commits by mistake while still allowing force-pushes. Even if it doesn't make sense from a security standpoint (main isn't really protected if one can still merge things via PRs), it would be really useful to prevent direct pushes done by mistake using the git command line.

**Comment by jsoref** [comment]
Personally, I use two different approaches for this:
* administrators can force push
* temporarily adjust the branch protection rule

I don't think that `git` itself exposes `force` as a thing that the server would understand. I believe that `force` is just a client flag that results in it the client being willing to submit changes that it would otherwise refuse to even offer.

**Reply by albertvaka** [reply]
I don't think this is a `git` issue. While disabling force-pushes has to be enforced at git level, force-merges can be checked at the PR level.

**Reply by jsoref** [reply]
Nevermind, I checked the implementation, apparently force is a wire level flag.
